src/ctapipe/tools/optimize_event_selection.py

```diff
@@ -0,0 +1,233 @@
+    ).tag(config=True)
+
+    proton_file = traits.Path(
+        default_value=None,
+        directory_ok=False,
+        allow_none=True,
+        help=(
+            "Proton input filename and path. "
+            "Not needed, if ``optimization_algorithm = 'PercentileCuts'``."
+        ),
+    ).tag(config=True)
+
+    proton_target_spectrum = traits.UseEnum(
+        Spectra,
+        default_value=Spectra.IRFDOC_PROTON_SPECTRUM,
+        help="Name of the spectrum used for weights of proton events.",
+    ).tag(config=True)
+
+    electron_file = traits.Path(
+        default_value=None,
+        directory_ok=False,
+        allow_none=True,
+        help=(
+            "Electron input filename and path. "
+            "Not needed, if ``optimization_algorithm = 'PercentileCuts'``."
+        ),
+    ).tag(config=True)
+
+    electron_target_spectrum = traits.UseEnum(
+        Spectra,
+        default_value=Spectra.IRFDOC_ELECTRON_SPECTRUM,
+        help="Name of the spectrum used for weights of electron events.",
+    ).tag(config=True)
+
+    chunk_size = Integer(
+        default_value=100000,
+        allow_none=True,
+        help="How many subarray events to load at once when preselecting events.",
+    ).tag(config=True)
+
+    output_path = traits.Path(
+        default_value="./Selection_Cuts.fits",
+        allow_none=False,
+        directory_ok=False,
+        help="Output file storing optimization result",
+    ).tag(config=True)
+
+    obs_time = AstroQuantity(
+        default_value=u.Quantity(50, u.hour),
+        physical_type=u.physical.time,
+        help=(
+            "Observation time in the form ``<value> <unit>``."
+            " This is used for flux normalization when calculating sensitivities."
+        ),
+    ).tag(config=True)
+
+    optimization_algorithm = traits.ComponentName(
+        CutOptimizerBase,
+        default_value="PointSourceSensitivityOptimizer",
+        help="The cut optimization algorithm to be used.",
+    ).tag(config=True)
+
+    aliases = {
+        "gamma-file": "EventSelectionOptimizer.gamma_file",
+        "proton-file": "EventSelectionOptimizer.proton_file",
+        "electron-file": "EventSelectionOptimizer.electron_file",
+        "output": "EventSelectionOptimizer.output_path",
+        "chunk_size": "EventSelectionOptimizer.chunk_size",
+    }
+
+    classes = [EventLoader] + classes_with_traits(CutOptimizerBase)
+
+    def setup(self):
+        """
+        Initialize components from config.
+        """
+        self.optimizer = CutOptimizerBase.from_name(
+            self.optimization_algorithm, parent=self
+        )
+        self.particles = [
+            EventLoader(
+                parent=self,
+                kind="gammas",
+                file=self.gamma_file,
+                target_spectrum=self.gamma_target_spectrum,
+            )
+        ]
+        if self.optimization_algorithm != "PercentileCuts":
```

Comment:
Might be safer to use `isinstance()` here and elsewhere instead of a string compare to the class name, so that if we ever re-factor or rename classes, it gets updated correctly. But that would mean importing those sub-classes or the module, so not clear if that makes it messier.